editor/src/app/sites/settings/site-settings.state.ts

```diff
@@ -76,7 +78,15 @@ export class SiteSettingsState implements NgxsOnInit {
       value: action.payload[settingKey]
     };
 
-    return this.appStateService.sync('siteSettings', data).pipe(
+    const url = action.settingGroup + '/' + settingKey;
+    const fileUpload$ = data.value instanceof File ?
+      this.fileUploadService.upload(url, data.value).pipe(
+        map(fileUpload => ({ ...data, value: fileUpload['filename'] })))
```

Comment:
you can specify the response type in `fileUploadService` so you don't have to access filename through square-brackets.